### PR TITLE
interp: change API of Eval

### LINKED ISSUE #731: Redeclaring imports
I used to load multiple scripts using **Eval** method in version 0.8.5. They had same package name and shared some same imports. Now I'm getting 'xxx redeclared in this block' (xxx being the imported package name).
Not sure why is this invalid now as this feature allowed me to split sources across multiple files.

### LINKED ISSUE #778: redeclared struct
The following program `sample.go` triggers a panic:

yaegi version: v0.8.9 -  v0.8.12 

```go
package yaegi

import (
	"github.com/containous/yaegi/interp"
	"testing"
)

const src = `package main

import (
)

type GetHistoryLocRsp struct { 
    DailyItems []DailyItem  
}

type DailyItem struct { 
    Timestamp int64    
}
`

func TestM(tt *testing.T) {
	i := interp.New(interp.Options{})
	_, err := i.Eval(src)
	if err != nil {
		panic(err)
	}
}
```

Expected result:
```console
the code will run without panic

```

Got:
```console
panic：

panic: 10:6: DailyItem redeclared in this block [recovered]
	panic: 10:6: DailyItem redeclared in this block

goroutine 270 [running]:
testing.tRunner.func1.1(0x1b55740, 0xc00017bae0)
	/home/yuanyangen/.install/go/src/testing/testing.go:940 +0x4dc
testing.tRunner.func1(0xc0003cc900)
	/home/yuanyangen/.install/go/src/testing/testing.go:943 +0x317
panic(0x1b55740, 0xc00017bae0)
	/home/yuanyangen/.install/go/src/runtime/panic.go:975 +0x41b
code.byted.org/hotsoon/func_core/executor/yaegi.TestM(0xc0003cc900)
	/home/yuanyangen/HomeData/go/func/func_core/executor/yaegi/yaegi_raw_test.go:26 +0x10e
testing.tRunner(0xc0003cc900, 0x1c72670)
	/home/yuanyangen/.install/go/src/testing/testing.go:991 +0x18a
created by testing.(*T).Run
	/home/yuanyangen/.install/go/src/testing/testing.go:1042 +0x660
```

### LINKED ISSUE #789: Maybe infinite loop while looking for missing vendored dependencies
TODO(mpl): to be confirmed/described

Hint: when a plugin looks for its deps in the vendor dir, but they are absent.

### LINKED ISSUE #798: interp: another undetected redeclaration
```
mbp:foo mpl$ tree
.
├── foo.go
└── vendor
    └── guthib.com
        ├── bar
        │   └── quux.go
        └── baz
            └── quux.go
```

```go
$ cat vendor/guthib.com/bar/quux.go
package quux

func Quux() string {
	return "bar"
}
```

```go
$ cat vendor/guthib.com/baz/quux.go
package quux

func Quux() string {
	return "baz"
}
```

```go
$ cat foo.go
package main

import (
	"guthib.com/bar"
	"guthib.com/baz"
)

func main() {
	println("Hello", bar.Quux())
	println("Hello", baz.Quux())
}
```

Expected result:
```console
$ go run ./foo.go
# command-line-arguments
./foo.go:5:2: quux redeclared as imported package name
	previous declaration at ./foo.go:4:2
./foo.go:5:2: imported and not used: "foo/vendor/guthib.com/baz" as quux
./foo.go:9:19: undefined: bar
./foo.go:10:19: undefined: baz
```

Got:
```console
$ yaegi ./foo.go
Hello bar
Hello baz
```

### PR DESCRIPTION
**Pull request by mpl**
The recent changes that added some redeclaration checks implicitly added more
strictness related to namespaces and scopes which, among other things, broke
some uses that "accidentally" used to work.

For example, given

```
const script1 = `
	import "fmt"

	// more code
`
const script2 = `
	import "fmt"

	// some other code
`
```

If one Evals script1, then script2, with the same interpreter, without
specifying any scope, as the two fragments would be considered part of the same
(.go file) scope by default, a redeclaration error would be triggered because
`import "fmt"` is seen twice.

A work-around would have been to specify (a different) `i.Name` before each Eval
call, so that each script is considered as coming from a different .go file, and
hence are respectively in different scopes with respect to imports.

That lead us to realize we had to make specifying things such as file-scope, and
"incremental mode" (aka REPL), more obvious in the context of an Eval call.

In addition, we want to lay down the foundations for Yaegi being able to behave
more like the go tool wrt to various inputs, i.e. it should be able to take a
package directory, or an import path, as input, instead of just a .go file.

Hence the introduction of a new kind of Eval method (whose signature is not fixed yet):

```
func (interp *Interpreter) EvalPath(path string) (res reflect.Value, err error)
```

It partially solves the problem described above because:
1) the path given to EvalPath can be used as the file-scope hint mentionned
above, for now (even though the related implementation details might change).
2) Eval always runs in incremental mode, whereas EvalPath always runs in
non-incremental mode, hence clarifying the sitation in that respect.

And to avoid confusion, the `Name` field of Interpreter is now non-exported,
since it is somewhat redundant with the `path` argument of EvalPath.

Note that #731 is not fully fixed (and might never be), as a requirement of the
proposed solution is to move the input bits of code into respective files
(instead of leaving them as strings).

Finally, some related bugfixes, documention changes, and some refactoring have
been included. Notably, there is no "empty scope" anymore, i.e. `name` defaults
to "_.go" when it is not specified.

Updates #731
Fixes #778
Fixes #798 
Fixes #789

### CODE REVIEW

**Comment by mvertes** [comment]
After rebasing, and following discussions with @mpl, I also reverted the introduction of `EvalInc`, keeping `Eval` instead. I have also introduced `EvalName` and `EvalPath` public methods and `eval` private method. I suppressed the `inREPL` field in `Interpreter` struct.

**Review by mvertes — APPROVED** [review]
LGTM

File: interp/interp.go [hunk 1]
```diff
@@ -330,26 +329,38 @@ func (interp *Interpreter) main() *node {
...
 
-// Eval evaluates Go code represented as a string. It returns a map on
-// current interpreted package exported symbols.
-// Name is the file path to the Go source file containing src. It defaults to the
-// special value "_.go".
-// Incremental is whether the given source should be considered an integral
-// valid Go source file, or only a fragment.
-func (interp *Interpreter) Eval(src, name string, incremental bool) (res reflect.Value, err error) {
+// EvalName evaluates Go code represented as a string. Name is the import path of code
+// and can be left to "" for main. EvalName returns the last result computed by the
```
Comment (mpl) [inline interp/interp.go:339]: "can" or "should" ?

File: interp/interp.go [hunk 2]
```diff
@@ -330,26 +329,38 @@ func (interp *Interpreter) main() *node {
...
+// interpreter, and a non nil error in case of failure.
+func (interp *Interpreter) EvalName(src, name string) (res reflect.Value, err error) {
+	return interp.eval(src, name, true)
+}
+
+// EvalPath evaluates Go file by path. EvalPath returns the last result
```
Comment (mpl) [inline interp/interp.go:345]:
```suggestion
// EvalPath evaluates the Go code located at path. EvalPath returns the last result
```

File: interp/interp.go [hunk 3]
```diff
@@ -330,26 +329,38 @@ func (interp *Interpreter) main() *node {
...
-// Name is the file path to the Go source file containing src. It defaults to the
-// special value "_.go".
-// Incremental is whether the given source should be considered an integral
-// valid Go source file, or only a fragment.
-func (interp *Interpreter) Eval(src, name string, incremental bool) (res reflect.Value, err error) {
+// EvalName evaluates Go code represented as a string. Name is the import path of code
+// and can be left to "" for main. EvalName returns the last result computed by the
+// interpreter, and a non nil error in case of failure.
+func (interp *Interpreter) EvalName(src, name string) (res reflect.Value, err error) {
+	return interp.eval(src, name, true)
```
Comment (mpl) [inline interp/interp.go:342]: Why is EvalName always in incremental mode?
And if it is, I think it's weird that some of the Eval* methods are in incremental mode, whereas some others are not, and that we haven't even documented it for the user.

File: interp/src.go [hunk 4]
```diff
@@ -88,26 +100,27 @@ func (interp *Interpreter) importSrc(rPath, path string) (string, error) {
...
 		if err != nil {
 			return "", err
 		}
 		revisit[subRPath] = append(revisit[subRPath], list...)
 	}
 
 	// Revisit incomplete nodes where GTA could not complete.
-	for pkg, nodes := range revisit {
-		if err = interp.gtaRetry(nodes, pkg, path); err != nil {
+	//	for pkg, nodes := range revisit {
```
Comment (mpl) [inline interp/src.go:113]: TODO(mpl): remove

**Pull request merged**